notify/notify.go

```diff
@@ -476,10 +472,16 @@ func (n *DedupStage) Exec(ctx context.Context, alerts ...*types.Alert) (context.
 		return ctx, nil, fmt.Errorf("repeat interval missing")
 	}
 
-	hash := n.hash(alerts)
-	resolved := n.resolved(alerts)
+	firing := map[uint64]struct{}{}
+	resolved := map[uint64]struct{}{}
 
-	ctx = WithNotificationHash(ctx, hash)
```

Comment:
This context entry is completely gone now. The related function and const should be removed as well.

But we are redoing the whole hashing we have done before. Wouldn't it make sense to pass the hashes through the context to the SetNotifies stage?